krping.c

```diff
@@ -522,10 +522,14 @@ static int krping_setup_buffers(struct krping_cb *cb)
 	cb->recv_dma_addr = ib_dma_map_single(cb->pd->device,
 				   &cb->recv_buf, 
 				   sizeof(cb->recv_buf), DMA_BIDIRECTIONAL);
+	if (unlikely(ib_dma_mapping_error(cb->pd->device, cb->recv_dma_addr)))
+		DEBUG_LOG(PFX "recv_buf DMA map failed\n");
```

Comment:
i think you need to handle these errors and not just log something...  If the mapping fails, then krping should bail out.